test/recipes/30-test_evp_data/evppkey_rfc6979.txt

```diff
@@ -0,0 +1,2795 @@
+Output = 3081880242013E99020ABF5CEE7525D16B69B229652AB6BDF2AFFCAEF38773B4B7D08725F10CDB93482FDCC54EDCEE91ECA4166B2A7C6265EF0CE2BD7051B7CEF945BABD47EE6D024201FBD0013C674AA79CB39849527916CE301C66EA7CE8B80682786AD60F98F7E78A19CA69EFF5C57400E3B3A0AD66CE0978214D13BAF4E9AC60752F7B155E2DE4DCE3
+
+DigestVerify = SHA512
+Key = P-521_PUB
+Input = "test"
+Output = 3081880242013E99020ABF5CEE7525D16B69B229652AB6BDF2AFFCAEF38773B4B7D08725F10CDB93482FDCC54EDCEE91ECA4166B2A7C6265EF0CE2BD7051B7CEF945BABD47EE6D024201FBD0013C674AA79CB39849527916CE301C66EA7CE8B80682786AD60F98F7E78A19CA69EFF5C57400E3B3A0AD66CE0978214D13BAF4E9AC60752F7B155E2DE4DCE2
+Result = VERIFY_ERROR
+
+Title = RFC 6979 K-163 deterministic ECDSA tests
+
+PrivateKey=K-163_PRIV
+-----BEGIN PRIVATE KEY-----
+MDICAQAwEAYHKoZIzj0CAQYFK4EEAAEEGzAZAgEBBBSaTWeSKVp/cw/D8rScvA9i6GInLw==
+-----END PRIVATE KEY-----
+
+PublicKey=K-163_PUB
+-----BEGIN PUBLIC KEY-----
+MEAwEAYHKoZIzj0CAQYFK4EEAAEDLAAEB5ruCQ2wXsJS1ctEUvNWvhmKT/lvB4LiljTdyaMe9AOG
+6Ja6oYtTr6Wj
+-----END PUBLIC KEY-----
+
+PrivPubKeyPair=K-163_PRIV:K-163_PUB
+
+DigestSign = SHA1
+Key = K-163_PRIV
+Ctrl = ecdsa_nonce_type:deterministic
+Input = "sample"
+Output = 302E0215030C45B80BA0E1406C4EFBBB7000D6DE4FA465D5050215038D87DF89493522FC4CD7DE1553BD9DBBA2123011
+
+DigestVerify = SHA1
+Key = K-163_PUB
+Input = "sample"
+Output = 302E0215030C45B80BA0E1406C4EFBBB7000D6DE4FA465D5050215038D87DF89493522FC4CD7DE1553BD9DBBA2123011
+
+DigestVerify = SHA1
+Key = K-163_PUB
+Input = "sample"
+Output = 302E0215030C45B80BA0E1406C4EFBBB7000D6DE4FA465D5050215038D87DF89493522FC4CD7DE1553BD9DBBA2123010
+Result = VERIFY_ERROR
+
+DigestSign = SHA224
+Key = K-163_PRIV
+Ctrl = ecdsa_nonce_type:deterministic
+Input = "sample"
+Output = 302D0215038A2749F7EA13BD5DA0C76C842F512D5A65FFAF32021464F841F70112B793FD773F5606BFA5AC2A04C1E8
+
+DigestVerify = SHA224
+Key = K-163_PUB
+Input = "sample"
+Output = 302D0215038A2749F7EA13BD5DA0C76C842F512D5A65FFAF32021464F841F70112B793FD773F5606BFA5AC2A04C1E8
+
+DigestVerify = SHA224
+Key = K-163_PUB
+Input = "sample"
+Output = 302D0215038A2749F7EA13BD5DA0C76C842F512D5A65FFAF32021464F841F70112B793FD773F5606BFA5AC2A04C1E9
+Result = VERIFY_ERROR
+
+DigestSign = SHA256
+Key = K-163_PRIV
+Ctrl = ecdsa_nonce_type:deterministic
+Input = "sample"
+Output = 302E02150113A63990598A3828C407C0F4D2438D990DF99A7F021501313A2E03F5412DDB296A22E2C455335545672D9F
+
+DigestVerify = SHA256
+Key = K-163_PUB
+Input = "sample"
+Output = 302E02150113A63990598A3828C407C0F4D2438D990DF99A7F021501313A2E03F5412DDB296A22E2C455335545672D9F
+
+DigestVerify = SHA256
+Key = K-163_PUB
+Input = "sample"
+Output = 302E02150113A63990598A3828C407C0F4D2438D990DF99A7F021501313A2E03F5412DDB296A22E2C455335545672D9E
+Result = VERIFY_ERROR
+
+DigestSign = SHA384
+Key = K-163_PRIV
+Ctrl = ecdsa_nonce_type:deterministic
+Input = "sample"
+Output = 302E0215034D4DE955871BB84FEA4E7D068BA5E9A11BD8B6C4021502BAAF4D4FD57F175C405A2F39F9755D9045C820BD
+
+DigestVerify = SHA384
+Key = K-163_PUB
```

Comment:
Why is the ctrl only set on the DigestSign?

Reply:
@slontis You won't be able to know whether the signature is deterministic when you verify it.
There is no need to treat deterministic signatures differently when verifying.